runtime/src/iree/tooling/parameter_util.c

```diff
@@ -33,6 +39,36 @@ static void iree_file_contents_release_callback(
   iree_file_contents_free(file_contents);
 }
 
+// Legacy parameter file open path. We should be able to replace this usage with
+// iree_io_file_handle_t-based logic.
+static iree_status_t iree_io_open_parameter_file_legacy(
+    iree_string_view_t path, iree_file_read_flags_t read_flags,
+    iree_allocator_t host_allocator, iree_io_file_handle_t** out_file_handle) {
+  IREE_ASSERT_ARGUMENT(out_file_handle);
+  *out_file_handle = NULL;
+
+  char path_str[2048] = {0};
+  iree_string_view_to_cstring(path, path_str, sizeof(path_str));
```

Comment:
Same as other comments, if we do end up truncating we could open an unexpected file.